lib/ojs.mli

```diff
@@ -57,20 +57,38 @@ external fun_to_js_args: (t -> 'a) -> t = "caml_ojs_wrap_fun_arguments"
 
 (** {2 JS objects} *)
 
-external generic_get: t -> t -> t = "caml_js_get"
-
-external generic_set: t -> t -> t -> unit = "caml_js_set"
-
 external get: t -> string -> t = "caml_js_get"
+  [@@ocaml.deprecated "Use Ojs.get_prop_ascii instead."]
 
 external set: t -> string -> t -> unit = "caml_js_set"
+  [@@ocaml.deprecated "Use Ojs.set_prop_ascii instead."]
+
+external delete: t -> string -> unit = "caml_js_delete"
+  [@@ocaml.deprecated "Use Ojs.delete_prop_ascii instead."]
+
+external get_prop_ascii: t -> string -> t = "caml_js_get"
+  (** Get the property from an object (only works if the property key is a plain ascii string). *)
+
+external set_prop_ascii: t -> string -> t -> unit = "caml_js_set"
+  (** Set an object property (only works if the property key is a plain ascii string). *)
+
+external delete_prop_ascii: t -> string -> unit = "caml_js_delete"
+  (** Delete an object property (only works if the property key is a plain ascii string). *)
+
+external get_prop: t -> t -> t = "caml_js_get"
+  (** Get the property from an object. *)
+
+external set_prop: t -> t -> t -> unit = "caml_js_set"
+  (** Set an object property. *)
+
+external delete_prop: t -> t -> unit = "caml_js_delete"
+  (** Delete an object property. *)
 
 external obj: (string * t) array -> t = "caml_js_object"
 
 val empty_obj: unit -> t
 
 val has_property: t -> string -> bool
-external iter_properties: t -> (string -> unit) -> unit = "caml_ojs_iterate_properties"
```

Comment:
I removed this function because using it is usually a bad idea and it is buggy on UTF-8 strings (independently of the change in semantics).

Reply:
The function could be useful for debugging purposes, and to bind to JS objects that represent generic association lists.  What is the current implementation buggy?

Reply:
In particular, it goes through caml_js_to_string, which should be ok, no?

Reply:
> The function could be useful for debugging purposes, and to bind to JS objects that represent generic association lists. 

I'm ok with debugging purposes and super sceptic about using JS objects to represent generic association lists.

But Yes, there is not bug; I misread the implementation so I will revert this change and its consequences.